nucypher/blockchain/eth/utils.py

```diff
@@ -72,6 +73,23 @@ def datetime_to_period(datetime: maya.MayaDT) -> int:
     return int(future_period)
 
 
+def period_to_datetime(period: int) -> maya.MayaDT:
+    now = maya.now()
+    current_period = datetime_to_period(datetime=now)
+
+    delta_periods = period - current_period
+
+    # +
+    if delta_periods:
+        duration = now + maya.timedelta(days=delta_periods)
```

Comment:
`duration` is perhaps a misnomer here, no?

Isn't `datetime_at_period` better?  (though it's actual *this* time in the middle of *that* period, right?)  Or even simply `then`?

Reply:
Through working with @cygnusv and @michwill we uncovered some calculation issues in this function.  I will follow up with an RFC commit.

Reply:
:heavy_check_mark: